### PR TITLE
Remove Docker App docs

### PR DESCRIPTION
**Pull request by usha-mandya**
Signed-off-by: Usha Mandya <usha.mandya@docker.com>

Remove documentation on Docker App as it's no longer supported

### CODE REVIEW

File: _data/toc.yaml [hunk 1]
```diff
@@ -1284,8 +1284,6 @@ manuals:
   - path: /engine/deprecated/
     title: Deprecated features
 
-  - path: /app/working-with-app/
```
Comment (thaJeztah) [inline _data/toc.yaml:1287]: We should have a redirect to somewhere

There's also the reference docs (perhaps we need to keep them for now, but if we remove them, they need a redirect as well); https://github.com/docker/docker.github.io/tree/master/engine/reference/commandline

**Comment by usha-mandya** [comment]
@thaJeztah  @mat007 Here's a list of the updates PTAL

- Updated the yaml files to include deprecated notice
- Updated the deprecated notice to add a link to the Roadmap on deprecating Docker App
- Added a redirect and moved the content of the `Working with Docker App` topic to `/engine/reference/commandline/app/` page.
- The format of the extended description in `/engine/reference/commandline/app_completion/` needs to be fixed in the upstream repo. PS: This is currently broken on the prod site too - https://docs.docker.com/engine/reference/commandline/app_completion/

File: _data/toc.yaml (hunk 1 above)
Comment (usha-mandya) [reply _data/toc.yaml:1287]: This is fixed. Added a redirect and moved the content of the Working with Docker App topic to /engine/reference/commandline/app/ page.

File: _includes/cli.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ your client and daemon API versions.
 
 > This command is [deprecated](/engine/deprecated/){: target="_blank" rel="noopener" class="_"}.
 >
-> It may be removed in a future Docker version.
+> It may be removed in a future Docker version. For more information, see the [Docker Roadmap](https://github.com/docker/roadmap/issues/209){: target="_blank" rel="noopener" class="_"}.
```
Comment (mat007) [inline _includes/cli.md:23]: Oh good, we need to link to the roadmap from the next Desktop release notes as well.

**Review by mat007 — APPROVED** [review]
Makes sense!

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**